src/wireviz/wv_output.py

```diff
@@ -220,3 +218,35 @@ def generate_titlepage(yaml_data, extra_metadata, shared_bom, for_pdf=False):
         rendered={'index_table': index_table.render(options)},
         bom_render_options=bom_render_options,
     )
+
+
+def generate_xlsx_output(filename, inp, header=None):
+    print('Generating xlsx output')
+    workbook = xlsxwriter.Workbook(filename)
+    worksheet = workbook.add_worksheet()
+
+    populate_spreadsheet_header_row(worksheet, get_xlsx_row_ordered_dict(inp.entries[0]))
+    for row_idx, line_item in enumerate(inp.entries):
+        line_item = get_xlsx_row_ordered_dict(line_item)
+        for field in line_item:
+            worksheet.write(row_idx+1, list(line_item.keys()).index(field), line_item[field])
+
+        qty_to_buy_formula = f'= INDIRECT(Address({row_idx+1}, {get_excel_col_idx(line_item, "Qty")})) + INDIRECT(Address({row_idx+1}, {get_excel_col_idx(line_item, "Qty Extras Desired")})) - INDIRECT(Address({row_idx+1}, {get_excel_col_idx(line_item, "Qty Inventory")}))'#+1 because title row
```

Comment:
maybe it would be good to have a method of this kind:

```
def get_field_address(row_idx, line_item, column):
    return f'INDIRECT(Address({row_idx+1}, {get_excel_col_idx(line_item, "Qty")}))'
```

then:
```
qty_loc = get_field_address(row_idx, line_item, 'Qty')
extra_loc = get_field_address(row_idx, line_item, 'Qty Extras Desired')
inventory_loc = get_field_address(row_idx, line_item, 'Qty Inventory')
qty_to_buy_fomula = f'= {qty_loc} + {extra_loc} - {inventory_loc}'
```

This still hardcodes the name of the columns, which shouldn't be required

Reply:
Agreed, that's clearer